### PR TITLE
#608 When table is empty, show customizable text and Add button

### PR DESCRIPTION
**Pull request by nmgokhale**
Fixes https://github.com/elyra-ai/canvas/issues/608

Added empty table content for controls - 
- List
- Readonlytable
- Selectcolumns
- Structurelisteditor
- Structuretable
- Fieldpicker

Empty table content general format -
```
text: To begin, click "{button_label}"
button: tertiary button - label same as the Add button label in the table
```
- Text can be customized by the host applications for **all** controls by passing `"{parameter-id}.empty.table.text"` in resources in parameter definition.
For example - 
```
"resources": {
    "list_string.empty.table.text": "This is a custom empty text for list."
  }
```
- Button label will be different for different controls -
**Add value** - List, structurelisteditor
**Add columns** - Selectcolumns, Structuretable, Fieldpicker
**Edit** or any custom label (if defined) - Readonlytable

 List, structurelisteditor controls -
Screen Shot 2021-03-30 at 4 23 03 PM

Selectcolumns, Structuretable, Fieldpicker controls -
Screen Shot 2021-03-30 at 4 24 40 PM

Readonlytable control -
Screen Shot 2021-03-30 at 4 59 56 PM

Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: canvas_modules/common-canvas/src/common-properties/controls/abstract-table.jsx [hunk 1]
```diff
@@ -770,6 +771,35 @@ export default class AbstractTable extends React.Component {
 			content: subCell
 		};
 	}
+
+	makeEmptyTablePanel(emptyTableButtonLabel, buttonClickHandler, disabled) {
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/controls/abstract-table.jsx:775]: This almost seems like it's own react component.

File: canvas_modules/common-canvas/src/common-properties/controls/list/list.jsx [hunk 2]
```diff
@@ -158,19 +159,23 @@ class ListControl extends AbstractTable {
...
-				<MoveableTableRows
-					tableContainer={tableContainer}
-					control={this.props.control}
-					controller={this.props.controller}
-					propertyId={this.props.propertyId}
-					setScrollToRow={this.setScrollToRow}
-					setCurrentControlValueSelected={this.setCurrentControlValueSelected}
-					disabled={this.props.state === STATES.DISABLED}
-				/>
+				{isEmpty(this.props.value) && addRemoveRows ? this.makeEmptyTablePanel(tableButtonConfig.addButtonLabel, this.addRow, disabled)
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/controls/list/list.jsx:168]: Any reason not to put this logic all in MoveableTableRows?  Then each component doesn't need to have this check.

File: canvas_modules/common-canvas/src/common-properties/controls/selectcolumns/selectcolumns.jsx [hunk 3]
```diff
@@ -145,6 +146,8 @@ class SelectColumns extends AbstractTable {
 				<ValidationMessage state={this.props.state} messageInfo={this.props.messageInfo} />
 			</div>
 		);
+		const emptyTableButtonClickHandler = this.addOnClick.bind(this, this.props.propertyId);
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/controls/selectcolumns/selectcolumns.jsx:149]: I'm not sure if these 2 lines should be in the render method.  Shouldn't these be part of the constructor or componentDidMount?

File: canvas_modules/common-canvas/src/common-properties/controls/structuretable/structuretable.jsx [hunk 4]
```diff
@@ -168,6 +169,9 @@ class StructureTableControl extends AbstractTable {
 			</div>);
 
 		const onPanelContainer = this.getOnPanelContainer(this.props.selectedRows);
+		const emptyTableButtonClickHandler = this.addOnClick.bind(this, this.props.propertyId);
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/controls/structuretable/structuretable.jsx:172]: Same comment as above.

**Review by matthoward366 — COMMENTED** [review]
Looks good.  Just a couple comments.

**Review by caritaou — APPROVED** [review]

**Review by matthoward366 — APPROVED** [review]

**Pull request merged**